components/scream/src/dynamics/homme/physics_dynamics_remapper.hpp

```diff
@@ -684,7 +684,14 @@ do_remap_fwd() const
         EKAT_KERNEL_ERROR_MSG("Error! Unhandled case in switch statement.\n");
     }
   };
-  const auto policy = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(num_fields, 1);
+
+  const auto concurrency = KT::ExeSpace::concurrency();
+#ifdef KOKKOS_ENABLE_CUDA
+  const int team_size = std::min(1024, std::min(128*num_cols,32*(concurrency/nfields+31)/32));
+#else
+  const int team_size = (concurrency<num_fields ? 1 : concurrency/num_fields);
```

Comment:
If concurrency<num_fields (say, concurrency=6, and num_fields=7), a team size of 1 might not be great. When the last fields are processed (by a single thread), all other threads sit idle. Maybe we should set a minimum team size of 2, or 4?

Since we are planning on changing the team policy, it's not worth spending time on this though.